src/libcmd/installables.cc

```diff
@@ -615,42 +618,118 @@ std::tuple<std::string, FlakeRef, InstallableValue::DerivationInfo> InstallableF
 
     auto attrPath = attr->getAttrPathStr();
 
-    if (!attr->isDerivation())
-        throw Error("flake output attribute '%s' is not a derivation", attrPath);
+    auto type = attr->maybeGetStringAttr(state->sType).value_or("");
 
-    auto drvPath = attr->forceDerivation();
+    if (type == "derivation") {
 
-    std::set<std::string> outputsToInstall;
-    std::optional<NixInt> priority;
+        auto drvPath = attr->forceDerivation();
 
-    if (auto aMeta = attr->maybeGetAttr(state->sMeta)) {
-        if (auto aOutputsToInstall = aMeta->maybeGetAttr("outputsToInstall"))
-            for (auto & s : aOutputsToInstall->getListOfStrings())
-                outputsToInstall.insert(s);
-        if (auto aPriority = aMeta->maybeGetAttr("priority"))
-            priority = aPriority->getInt();
-    }
+        std::set<std::string> outputsToInstall;
+        std::optional<NixInt> priority;
+
+        if (auto aMeta = attr->maybeGetAttr(state->sMeta)) {
+            if (auto aOutputsToInstall = aMeta->maybeGetAttr("outputsToInstall"))
+                for (auto & s : aOutputsToInstall->getListOfStrings())
+                    outputsToInstall.insert(s);
+            if (auto aPriority = aMeta->maybeGetAttr("priority"))
+                priority = aPriority->getInt();
+        }
 
-    if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
-        outputsToInstall.clear();
-        if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
-            for (auto & s : aOutputs->getListOfStrings())
-                outputsToInstall.insert(s);
+        if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
+            outputsToInstall.clear();
+            if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
+                for (auto & s : aOutputs->getListOfStrings())
+                    outputsToInstall.insert(s);
+        }
+
+        if (outputsToInstall.empty())
+            outputsToInstall.insert("out");
+
+        if (auto outputNames = std::get_if<OutputNames>(&outputsSpec))
+            outputsToInstall = *outputNames;
+
+        auto drvInfo = DerivationInfo {
+            .drvPath = std::move(drvPath),
+            .outputsToInstall = std::move(outputsToInstall),
+            .priority = priority,
+        };
+
+        return {attrPath, getLockedFlake()->flake.lockedRef, std::move(drvInfo)};
     }
 
-    if (outputsToInstall.empty())
-        outputsToInstall.insert("out");
+    else if (type == "configurable") {
```

Comment:
If we want other things besides drvs/installables to be configurable, should we allow `type` to be some kind of structured value? E.g. `[ "configurable" "derivation" ]` or `{ configurable = true; configured = "derivation"; }` or something?

The alternative is to dynamically resolve this, i.e. check `type` after resolving the `configurable`.